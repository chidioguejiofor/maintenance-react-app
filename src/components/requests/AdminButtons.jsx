import React from 'react';
import { Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { getColor } from '../../helpers/requestHelper';

const AdminButtons = ({ status, onClick, requestId }) => {
  if (status === 'pending') {
    return (
      <span>
        <Button
          content="Approve"
          onClick={() => onClick('approve', requestId)}
          color={getColor('approved')}
          size="small"
        />
        <Button
          content="Disapprove"
          onClick={() => onClick('disapprove', requestId)}
          color={getColor('disapproved')}
          size="small"
        />
      </span>

    );
  }
  if (status === 'approved') {
    return (
      <span>
        <Button
          content="Resolve"
          onClick={() => onClick('resolve', requestId)}
          color={getColor('resolved')}
          size="small"
        />
        <Button
          content="Cancel"
          onClick={() => onClick('cancel', requestId)}
          color={getColor('disapproved')}
          size="small"
        />
      </span>

    );
  }

  return (
    <span>
      <Button
        content="Re-open"
        onClick={() => onClick('reopen', requestId)}
        color="brown"
        size="small"
      />
    </span>
  );
};

AdminButtons.propTypes = {
  status: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  requestId: PropTypes.number.isRequired,
};


export default AdminButtons;
