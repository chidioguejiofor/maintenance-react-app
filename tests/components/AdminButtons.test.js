import React from 'react';
import { shallow } from 'enzyme';
import chai from 'chai';
import { Button } from 'semantic-ui-react';
import AdminButtons from '../../src/components/requests/AdminButtons';

const sampleProps = {
  status: 'pending',
  onClick: () => {},
  requestId: 0
};
const getShallowObj = props => shallow(<AdminButtons {...sampleProps} {...props} />);
describe('Testing AdminButtons', () => {
  describe('components rendered', () => {
    it('should match the existing snapshot of the app ', () => {
      const wrapper = getShallowObj();
      expect(wrapper).toMatchSnapshot();
    });

    describe('when status is pending', () => {
      it('should render an approve and diaspprove buttons when status is "pending"', () => {
        const wrapper = getShallowObj({ status: 'pending' });
        chai.expect(wrapper.find(Button).length)
          .to.equal(2);
      });
      it('should call approve when the first button is clicked', () => {
        const mockClick = jest.fn();
        const wrapper = getShallowObj({ status: 'pending', onClick: mockClick });
        wrapper.find(Button).first().simulate('click');
        wrapper.find(Button).last().simulate('click');
        expect(mockClick)
          .toHaveBeenCalledTimes(2);
      });
    });

    describe('when status is "disapproved"', () => {
      it('should render an approve and disapproved buttons when status is "disapproved"', () => {
        const wrapper = getShallowObj({ status: 'disapproved' });
        chai.expect(wrapper.find(Button).length)
          .to.equal(1);
      });
      it('should call approve when the first button is clicked', () => {
        const mockClick = jest.fn();
        const wrapper = getShallowObj({ status: 'disapproved', onClick: mockClick });
        wrapper.find(Button).first().simulate('click');
        expect(mockClick)
          .toHaveBeenCalledTimes(1);
        expect(mockClick)
          .toHaveBeenCalledWith('approve', 0);
      });
    });
    describe('when status is "approved"', () => {
      it('should render an approve and disapproved buttons when status is "approved"', () => {
        const wrapper = getShallowObj({ status: 'approved' });
        chai.expect(wrapper.find(Button).length)
          .to.equal(2);
      });
      it('should call approve when the first button is clicked', () => {
        const mockClick = jest.fn();
        const wrapper = getShallowObj({ status: 'approved', onClick: mockClick });
        wrapper.find(Button).first().simulate('click');
        wrapper.find(Button).last().simulate('click');
        expect(mockClick)
          .toHaveBeenCalledTimes(2);
        expect(mockClick)
          .toHaveBeenCalledWith('resolve', 0);
        expect(mockClick)
          .toHaveBeenCalledWith('disapprove', 0);
      });
    });

    describe('when status is unknown', () => {
      it('should render no buttons when  "approved"', () => {
        const wrapper = getShallowObj({ status: 'unknown' });
        chai.expect(wrapper.find(Button).length)
          .to.equal(0);
      });
    });
  });
});
