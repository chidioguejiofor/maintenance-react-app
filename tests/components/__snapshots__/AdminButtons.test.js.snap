// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AdminButtons components rendered should match the existing snapshot of the app  1`] = `
<span>
  <Button
    as="button"
    color="green"
    content="Approve"
    onClick={[Function]}
    role="button"
    size="small"
  />
  <Button
    as="button"
    color="red"
    content="Disapprove"
    onClick={[Function]}
    role="button"
    size="small"
  />
</span>
`;
